### PR TITLE
fix line number off-by-one-error

### PR DESCRIPTION
**Pull request by asavahista**
**warning:** not extensively tested. if others could also test that'd be great.

please let me know if i need to add regression tests (or anything else)

it's a very small and simple change so just committing to master and closing this also works

### CODE REVIEW

**Comment by asavahista** [comment]
will close #730

**Comment by asavahista** [comment]
fixed the one test that was affected; the failing ci runs are the usual debian 4.02 to 4.05 - i'm assuming `with type foo := bar` in that one failing test was introduced in 4.06? if this is something that can be selectively disabled then someone might want to do that

File: test/xref2/references_to_pages.t/run.t [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # References to pages and items in pages
 
   $ compile p.mld good_references.mli bad_references.mli
-  File "p.mld", line 6, characters 5-11:
+  File "p.mld", line 7, characters 5-11:
```
Comment (Julow) [inline test/xref2/references_to_pages.t/run.t:4]: This test is wrong, these references should be changed to: (the errors will disappear)
```
{!Good_references} {!Good_references.t}
```

**Review by Julow — APPROVED** [review]
LGTM !

Adding a test would be nice. The place to do that would be `test/integration/compile.t/`, you'd need to add an `.mld` file with an error (eg. not beginning with a heading) and a `odoc compile` command in `run.t`.

**Comment by asavahista** [comment]
seems like not starting with a heading makes it stop completely (and not show any line numbers) - i'll just use some other error instead then

File: test/xref2/references_to_pages.t/run.t (hunk 1 above)
Comment (Julow) [reply test/xref2/references_to_pages.t/run.t]: This test and the other tests failing in CI will be fixed in https://github.com/ocaml/odoc/pull/738

**Comment by asavahista** [comment]
(hopefully) removed the fix for the unrelated test which is now in #738

**Comment by jonludlam** [comment]
Slightly surprisingly, the [examples for odoc-parser](https://ocaml-doc.github.io/odoc-parser/#example-usage) got this right. Surprising because the code was extracted from here :-)

**Pull request merged**